server/stream.go

```diff
@@ -5961,10 +5946,9 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 
 	if err != nil {
 		if isPermissionError(err) {
-			mset.mu.Unlock()
 			// messages in block cache could be lost in the worst case.
 			// In the clustered mode it is very highly unlikely as a result of replication.
-			mset.srv.DisableJetStream()
+			go mset.srv.DisableJetStream()
```

Comment:
Is this safe in a goroutine?

Reply:
Currently we would deadlock, because we're already holding the stream lock for this specific stream. So when disabling JetStream inline like this, we'll block on this stream lock while disabling.